swim/suspicion.go

```diff
@@ -33,62 +33,85 @@ type suspect interface {
 	incarnation() int64
 }
 
+type timer struct {
+	*time.Timer
+	// at which state the timer was started
+	state string
+}
+
 // Suspicion handles the suspicion sub-protocol of the SWIM protocol
 type suspicion struct {
```

Comment:
Can we come up with a better name for this struct if it will be doing more than the suspicion timer? First that comes to mind is 'delayed state change' but that is a long word. Anyone having a better suggestion?